### PR TITLE
Use Java 11 methods where possible

### PR DESCRIPTION
**Pull request by basil**
Applies the OpenRewrite [Migrate to Java 11 from Java 8](https://docs.openrewrite.org/tutorials/migrate-from-java-8-to-java-11) migration against this code base, along with any suggestions from IntelliJ's Java language migration aids and some manual touch-up.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6700"></a>

### CODE REVIEW

File: bom/pom.xml
Comment (basil) [inline bom/pom.xml:44]: https://github.com/jenkinsci/stapler/pull/377

File: pom.xml
Comment (basil) [inline pom.xml:92]: https://github.com/jenkinsci/remoting/pull/551

**Review by jglick — COMMENTED** [review]
Other than the POM bumps, all looks fine to merge right away, from a random sampling of diffs (did not attempt to read every line). Any reason to hold back on this?

**Comment by basil** [comment]
> Any reason to hold back on this?

Testing is still underway in https://github.com/jenkinsci/bom/pull/1234

**Comment by basil** [comment]
BOM test passed: https://ci.jenkins.io/job/Tools/job/bom/job/PR-1234/5/

File: pom.xml
Comment (basil) [reply pom.xml]: Correct, have reverted now that BOM tests passed

**Comment by basil** [comment]
> did not attempt to read every line)

I suggest at least reading the changes to `core/src/main/resources/jenkins/security/whitelisted-classes.txt` which were needed to prevent numerous failures in the core test suite

**Review by timja — APPROVED** [review]
I looked through most of it, nice job :)

**Review by NotMyFault — APPROVED** [review]
Hot stuff, nice one 🔥

**Review by StefanSpieker — APPROVED** [review]

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

File: core/src/main/resources/jenkins/security/whitelisted-classes.txt
Comment (jglick) [inline core/src/main/resources/jenkins/security/whitelisted-classes.txt:61]: (private top-level class in `ImmutableCollections.java`)

File: core/src/main/resources/jenkins/security/whitelisted-classes.txt
Comment (jglick) [inline core/src/main/resources/jenkins/security/whitelisted-classes.txt:107]: Just saw https://github.com/jenkinsci/jenkins/pull/6700#issuecomment-1169217726. For reference, I think this is so that tests like https://github.com/jenkinsci/jenkins/blob/a6ec1b024a9b6136677a3cb7505fdb690001bbac/test/src/test/java/hudson/model/ParametersAction2Test.java#L131 could still work. We would not expect such entries to be used in production code, any more than we would older entries like https://github.com/jenkinsci/jenkins/blob/a6ec1b024a9b6136677a3cb7505fdb690001bbac/core/src/main/resources/jenkins/security/whitelisted-classes.txt#L78

File: core/src/main/java/hudson/Functions.java
Comment (jglick) [inline core/src/main/java/hudson/Functions.java:1940]: Incompatible I think: https://github.com/jenkinsci/emailext-template-plugin/pull/71#issuecomment-1261157790

**Pull request merged**
